Objects/typeobject.c

```diff
@@ -3066,7 +3070,11 @@ PyType_FromSpecWithBases(PyType_Spec *spec, PyObject *bases)
             *
             * We store the user-provided traverse function at the end of the type
             * (we have allocated space for it) so we can call it from our
-            * PyType_FromSpec_tp_traverse wrapper. */
+            * PyType_FromSpec_tp_traverse wrapper.
+            *
+            * Check bpo-40217 for more information and rationale about this issue.
```

Comment:
Your phrasing is good and nice. But my suggestion was just to add "bpo-40217: " at the beginning of the comment. It's up to you.

Reply:
I personally like this way a bit more, bit I can move it to the top of you want